### PR TITLE
Rename existing aperture functions.

### PR DESCRIPTION
**Pull request by ehpor**
I deprecated the old aperture function names `circular_aperture()` and others, to unify the naming scheme between all aperture creation functions to `make_*_aperture()`.

The deprecated functions are still available, but will spit out a DeprecationWarning upon use. These decorators are now included for more general use in HCIPy as well, for future deprecations (instead of doing this manually, as I used to do).

This is a draft pull request because of this *question:* should we change this to `make_circle()` and similar names and drop the "aperture" as well?

### CODE REVIEW

File: hcipy/aperture/generic.py
Comment (syhaffert) [inline hcipy/aperture/generic.py:519]: Do we want to make them deprecated immediately? Or what until the next major tick?

**Review by syhaffert — DISMISSED** [review]

File: hcipy/aperture/generic.py
Comment (syhaffert) [reply hcipy/aperture/generic.py:519]: Never mind this comment. I read the new decorator code and I approve.

**Comment by ehpor** [comment]
@syhaffert @dsdoelman You also agree on not renaming these `make_circle()` and similar names? I'd rather not go through name changes twice for no reason. (Tagging David, cuz he reminded/strongly suggested me to make this change, finally.) I'd like both of your responses before merging.

File: hcipy/aperture/generic.py
Comment (ehpor) [reply hcipy/aperture/generic.py:519]: I'm aiming to have them deprecated for at least one minor release *and* for at least a year. I unofficially do (and have done) this for all breaking changes.

**Comment by dsdoelman** [comment]
I like the 'aperture'. I would never change functions like 'make_keck_aperture()'  -> 'make_keck()', so why change the circular aperture? I know people use it to create all kind of shapes for focal plane masks, but perhaps we can just make both and point them to the same function?

**Comment by syhaffert** [comment]
I think we should keep aperture. This has been the style since the beginning. And in the focal plane a mask is still an aperture, just in a different plane. But we could always add another function for overloading if people think it is too much typing.

**Comment by ehpor** [comment]
Okay, Thanks to you both! I'll keep these names then. It's easy to add aliases in the future if we desire that. Full name changes was what I was worried about. Merging this after #140.

**Review by syhaffert — APPROVED** [review]
Looks good to me.

**Comment by ehpor** [comment]
Actually, I forgot to rebase to fix the deprecation warnings due to #140 (https://github.com/ehpor/hcipy/pull/140#discussion_r948010783 and https://github.com/ehpor/hcipy/pull/140#discussion_r948011154). Let me do that, and then I need another approval @syhaffert . Sorry for that.

**Comment by ehpor** [comment]
Done. @syhaffert

**Review by syhaffert — APPROVED** [review]
looks good to merge.

**Pull request merged**
